client/src/endpoint/orderbook.rs

```diff
@@ -0,0 +1,127 @@
+//! Contains the endpoint for fetching the orderbook for a given asset pair
+use error::Result;
+use std::str::FromStr;
+use stellar_resources::{AssetIdentifier, Orderbook};
+use super::{Body, IntoRequest, Limit};
+use http::{Request, Uri};
+
+/// Given an asset pair, the endpoint will return all bids and asks with an optional
+/// limit parameter to cap the number of records returned.
+///
+/// <https://www.stellar.org/developers/horizon/reference/endpoints/orderbook-details.html>
+///
+/// ## Example
+/// ```
+/// use stellar_client::sync::Client;
+/// use stellar_client::endpoint::{orderbook, trades, Limit};
+///
+/// let client = Client::horizon_test().unwrap();
+///
+/// // Grab a trade so we can get two valid assets.
+/// # let endpoint = trades::All::default().with_limit(1);
+/// # let records = client.request(endpoint).unwrap();
+/// # let trade = &records.records()[0];
+/// # let asset1 = trade.base_asset().clone();
+/// # let asset2 = trade.counter_asset().clone();
+/// let orderbook_ep = orderbook::Details::for_asset_pair(asset1, asset2);
+/// let orderbook   = client.request(orderbook_ep).unwrap();
+///
+/// assert_eq!(orderbook.base(), trade.base_asset());
+/// ```
+#[derive(Debug, Limit)]
+pub struct Details {
+    base_asset: AssetIdentifier,
+    counter_asset: AssetIdentifier,
+    limit: Option<u32>,
+}
+
+impl Details {
+    /// Creates a new orderbook::Details endpoint struct. Hand this to the client in order to request
+    /// information about all bids and asks for an asset pair.
+    ///
+    /// ```
+    /// # extern crate stellar_client;
+    /// # extern crate stellar_resources;
+    ///
+    /// use stellar_client::endpoint::orderbook;
+    /// use stellar_resources::AssetIdentifier;
+    ///
+    /// # fn main() {
+    ///     let base = AssetIdentifier::native();
```

Comment:
we should remove the indentation here since it will look funny in the documentation.